src/backend/backend.did

```diff
@@ -71,22 +75,28 @@ type CreateDeploymentResult = variant {
 };
 
 //// IC WebSocket types ////
+import "./ws_types.did";
 
-// TODO: add IC WebSocket types
-
+type DeploymentUpdateWsMessage = record {
+    id : text;
+    update : DeploymentUpdate;
+};
 //// End IC WebSocket types ////
 
 service : (bool) -> {
+    "address" : () -> (ApiStringResult);
+    "balance" : () -> (ApiStringResult);
     "get_user" : () -> (GetUserResult) query;
     "create_user" : () -> (CreateUserResult);
     "promote_user_to_admin" : (UserId) -> (EmptyResult);
     "get_deployment" : (text) -> (GetDeploymentResult) query;
     "get_deployments" : () -> (GetDeploymentsResult) query;
     "create_deployment" : (text) -> (CreateDeploymentResult);
+    "create_test_deployment" : () -> (CreateDeploymentResult);
 
     //// IC WebSocket methods ////
-
-    // TODO: add IC WebSocket methods
-
+    "ws_open" : (CanisterWsOpenArguments) -> (CanisterWsOpenResult);
+    "ws_close" : (CanisterWsCloseArguments) -> (CanisterWsCloseResult);
+    "ws_get_messages" : (CanisterWsGetMessagesArguments) -> (CanisterWsGetMessagesResult) query;
     //// End IC WebSocket methods ////
```

Comment:
Do we need it even if we do not send messages?

Reply:
From the client I mean

Reply:
I wouldn't never remove ws protocol methods just because we don't need them (same reason why we've declared it in the endpoints).
Plus, the `ws_message` method in the candid is the most important, because it allows the ws client library to automatically generate the correct types for the messages.